vfs/http_streambuf.cc

```diff
@@ -140,6 +148,8 @@ struct HttpStreamingDownloadSource {
 
         bool httpAbortOnSlowConnection;
 
+        shared_ptr<FsObjectInfo> info;
```

Comment:
Instead of info, I would simply store a std::exception_ptr.  Use std::get_exception()